### PR TITLE
UIQM-716: Consolidate bib and authority routes to avoid page refresh after redirecting from the create/derive page to the edit one.

### PR DESCRIPTION
**Pull request by Dmytro-Melnyshyn**
## Purpose
Add a "Save & keep editing" button to the bibliographic and authority record creation and deriving pages. After hitting the button, redirect to the editing page without refreshing it and retain the focus.

## Approach
- consolidate bibliographic and authority routes to avoid refreshing the page;
- delete `QuickMarcCreateWrapper.js`, `QuickMarcEdiitWrapper.js` and `QuickMarcDeriveWrapper.js`. Their functionality has been carried over to a new `useSaveRecord` hook. They use the same UI that should be used directly in `QuickMarcEditorContainer` to avoid remounting fields after redirecting to the edit page, which prevents having the same ref for focusing;
- add a "Save & keep editing" button for bibliographic and authority records, update page data as soon as the record is created/derived, and then redirect to the edit page.

## Issues
[UIQM-716](https://folio-org.atlassian.net/browse/UIQM-716)

## Screencasts

https://github.com/user-attachments/assets/e871ab88-bde4-42e3-a73b-c8114f869e23

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by BogdanDenis** [comment]
@Dmytro-Melnyshyn ok let's continue with your approach.
Just having separate hooks for Create, Edit and Derive won't be enough, because they'd have to be still called in one place and then conditionally pass the results to `<QuickMarcEditor>`.
Instead maybe we could pass these hooks as props, just like the wrappers
Or alternatively, use a component factory to pass a hook like a regular function argument
Here's an article with some details and examples https://careers.wolt.com/en/blog/engineering/injecting-hooks-into-react-components , "Passing Hooks in Props" section
So, three separate hooks `useMarcCreate`, `useMarcEdit` and `useMarcDerive` and pass one of them as props or via component factory approach to `<QuickMarcEditorContainer>`. Then `<QuickMarcEditorContainer>` will call a hook and pass the results to `<QuickMarcEditor>`

**Comment by Dmytro-Melnyshyn** [comment]
@BogdanDenis, it doesn't work, React detects the order of hooks and crashes the page. The `useMarcEdit` contains `useMarcRecordMutation`, but `useMarcCreate` doesn't (React doesn't crash the page if the only difference is in hooks like `history`, `useParams`, ...).

I get the following error:
```
StripesContext.js:11 Warning: React has detected a change in the order of Hooks called by QuickMarcEditorContainer. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks
```

**Review by BogdanDenis — APPROVED** [review]

File: src/QuickMarc.js [hunk 1]
```diff
@@ -24,40 +20,41 @@ const QuickMarc = ({
...
   onSave,
 }) => {
+  const location = useLocation();
+
+  const permissionsMap = {
+    'create-bibliographic': 'ui-quick-marc.quick-marc-editor.create',
+    'edit-bibliographic': 'ui-quick-marc.quick-marc-editor.all',
+    'derive-bibliographic': 'ui-quick-marc.quick-marc-editor.derive.execute',
+    'create-authority': 'ui-quick-marc.quick-marc-authorities-editor.create',
+    'edit-authority': '', // ui-quick-marc.quick-marc-authorities-editor.all
```
Comment (Terala-Priyanka) [inline src/QuickMarc.js:30]: is this comment intentional?

Comment (Dmytro-Melnyshyn) [reply src/QuickMarc.js:30]: This permission was commented in the previous implementation, and I decided to leave it too.

**Review by Terala-Priyanka — APPROVED** [review]

**Pull request merged**